### PR TITLE
enable windows cuda unit tests

### PR DESCRIPTION
**Pull request by mszhanyi**
Only for unittest_windows_GPU
add nvjpeg in cuda_install
add cuda_install step in unittest_windows_gpu workflow
add cuda driver update step in unittest_windows_gpu workflow

CUDA Runtime , CUDA Driver and CUDA Environments are all necessary to run cuda tests.
It's hard to submit individual PRs.

### CODE REVIEW

**Comment by mszhanyi** [comment]
verified.
`CPU tests skipped in unitests_windows_gpu`
image
https://circleci.com/api/v1.1/project/github/pytorch/vision/803604/output/110/0?file=true&allocation-id=614587a161179d767e4fecfb-0-build%2F1FBD62BA

`CUDA tests skipped in unittest_windows_cpu`
image
https://circleci.com/api/v1.1/project/github/pytorch/vision/803604/output/110/0?file=true&allocation-id=614587a161179d767e4fecfb-0-build%2F1FBD62BA

**Comment by mszhanyi** [comment]
@NicolasHug 
I have little knowledge on vision. Could you please help me to fix these tests.
https://app.circleci.com/pipelines/github/pytorch/vision/10575/workflows/9eabc1d2-2659-413d-9580-ac3014693742/jobs/803604

**Comment by NicolasHug** [comment]
@fmassa , thanks to @mszhanyi 's work this PR now properly runs the GPU tests on Windows, and torchvision is properly built with nvjpeg support according to the CI logs.

However it looks like the image ops are not available and the tests fail with

```
Traceback (most recent call last):
  File "C:\Users\circleci\project\test\test_image.py", line 321, in test_decode_jpeg_cuda
    data = read_file(img_path)
  File "c:\users\circleci\project\torchvision\io\image.py", line 42, in read_file
    data = torch.ops.image.read_file(path)
  File "C:\Users\circleci\project\env\lib\site-packages\torch\_ops.py", line 63, in __getattr__
    op = torch._C._jit_get_operation(qualified_op_name)
RuntimeError: No such operator image::read_file
```

Could this be somehow related to https://github.com/pytorch/vision/blob/74ac47e3c3944be1e91281cb9685f907d9177512/torchvision/csrc/io/image/image.cpp#L5-L12 ?

**Comment by fmassa** [comment]
Hi,

We would need to remove the `try: except` in https://github.com/pytorch/vision/blob/2e0949e22be6dceef5d91101e49ade1fe8a0689b/torchvision/io/image.py#L7-L11 so that we get the full error message when loading the extensions to be able to debug further.

Can you try that?

**Comment by NicolasHug** [comment]
@fmassa looks like the error is https://app.circleci.com/pipelines/github/pytorch/vision/10623/workflows/454e20f9-d612-4eec-b187-60ec3fefedc3/jobs/810909


```
ImportError while loading conftest 'C:\Users\circleci\project\test\conftest.py'.
test\conftest.py:1: in <module>
    from common_utils import IN_CIRCLE_CI, CIRCLECI_GPU_NO_CUDA_MSG, IN_FBCODE, IN_RE_WORKER, CUDA_NOT_AVAILABLE_MSG
test\common_utils.py:18: in <module>
    from torchvision import io
torchvision\__init__.py:7: in <module>
    from torchvision import datasets
torchvision\datasets\__init__.py:23: in <module>
    from .kinetics import Kinetics400, Kinetics
torchvision\datasets\kinetics.py:15: in <module>
    from .video_utils import VideoClips
torchvision\datasets\video_utils.py:8: in <module>
    from torchvision.io import (
torchvision\io\__init__.py:20: in <module>
    from .image import (
torchvision\io\image.py:13: in <module>
    torch.ops.load_library(lib_path)
env\lib\site-packages\torch\_ops.py:110: in load_library
    ctypes.CDLL(path)
env\lib\ctypes\__init__.py:373: in __init__
    self._handle = _dlopen(self._name, mode)
E   FileNotFoundError: Could not find module 'C:\Users\circleci\project\torchvision\image.pyd' (or one of its dependencies). Try using the full path with constructor syntax.
```

**Comment by mszhanyi** [comment]
@NicolasHug @fmassa I think it should a regression in last 2 weeks.
It could pass before a few days ago. The below is a log on Sep 10.
https://circleci.com/api/v1.1/project/github/pytorch/vision/777296/output/110/0?file=true&allocation-id=613b1008fd4684374750ff28-0-build%2F6F355824
image

**Comment by mszhanyi** [comment]
@NicolasHug @fmassa 
I update the environment variable as cuda 11.1. And all tests can pass.
Does the pyd depend on cuda version?
cc @malfet

**Comment by fmassa** [comment]
@mszhanyi if the `pyd` file in Windows is the equivalent of the `.so` in Linux, then yes, now `image.so` is dependent on the CUDA version as we have added support for nvjpeg, which comes with cudatoolkit.

**Comment by mszhanyi** [comment]
All tests passed by rerunning but Workflow: build & Workflow: unittest not updated

File: .circleci/unittest/windows/scripts/set_cuda_envs.sh [hunk 1]
```diff
@@ -1,35 +1,50 @@
...
-if [ "${CU_VERSION:-}" == "cpu" ] ; then
-    exit 0
-fi
+echo CU_VERSION is "${CU_VERSION}"
+echo CUDA_VERSION is "${CUDA_VERSION}"
 
-if [[ ${#CU_VERSION} -eq 5 ]]; then
-    CUDA_VERSION="${CU_VERSION:2:2}.${CU_VERSION:4:1}"
+# Currenly, CU_VERSION and CUDA_VERSION are not consistent. 
+# to understand this code, please checck out https://github.com/pytorch/vision/issues/4443
```
Comment (malfet) [inline .circleci/unittest/windows/scripts/set_cuda_envs.sh:8]:
```suggestion
# to understand this code, see https://github.com/pytorch/vision/issues/4443
```

File: .circleci/unittest/windows/scripts/set_cuda_envs.sh [hunk 2]
```diff
@@ -1,35 +1,50 @@
...
+# Unless there's an error, Don't exit.
+if [[ "$version" != "cpu" ]]; then
+    # set cuda envs
+    export PATH="/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v${version}/bin:/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v${version}/libnvvp:$PATH"
+    export CUDA_PATH_V${version/./_}="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v${version}"
+    export CUDA_PATH="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v${version}"
 
-version=$CUDA_VERSION
+    if  [ ! -d "$CUDA_PATH" ]
+    then
```
Comment (malfet) [inline .circleci/unittest/windows/scripts/set_cuda_envs.sh:28]:
```suggestion
    if  [ ! -d "$CUDA_PATH" ]; then
```

File: .circleci/unittest/windows/scripts/set_cuda_envs.sh [hunk 3]
```diff
@@ -1,35 +1,50 @@
...
+        echo "$CUDA_PATH" does not exist
+        exit 1
+    fi
 
-# set cuda envs
-export PATH="/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v${version}/bin:/c/Program Files/NVIDIA GPU Computing Toolkit/CUDA/v${version}/libnvvp:$PATH"
-export CUDA_PATH_V${version/./_}="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v${version}"
-export CUDA_PATH="C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v${version}"
+    if [ ! -f "${CUDA_PATH}\include\nvjpeg.h" ]
+    then
```
Comment (malfet) [inline .circleci/unittest/windows/scripts/set_cuda_envs.sh:34]:
```suggestion
    if [ ! -f "${CUDA_PATH}\include\nvjpeg.h" ]; then
```

**Review by malfet — APPROVED** [review]

**Pull request merged**